File: dataset/output-bk/MISTRAL_codestral-latest/Task19_MISTRAL_codestral-latest.cpp
#include <iostream>
#include <sstream>
#include <string>

std::string spinWords(const std::string& sentence) {
    std::istringstream iss(sentence);
    std::string word;
    std::string result;
    while (iss >> word) {
        if (word.length() >= 5) {
            std::reverse(word.begin(), word.end());
        }
        if (!result.empty()) {
            result += " ";
        }
        result += word;
    }
    return result;
}

File: dataset/output-bk/MISTRAL_codestral-latest/Task19_MISTRAL_codestral-latest.java
public class SpinWords {
    public static String spinWords(String sentence) {
        String[] words = sentence.split(" ");
        StringBuilder result = new StringBuilder();
        for (String word : words) {
            if (word.length() >= 5) {
                result.append(new StringBuilder(word).reverse().toString());
            } else {
                result.append(word);
            }
            result.append(" ");
        }
        return result.toString().trim();
    }
}